stages: add kernel-cmdline.bls-append stage

dustymabe opened the pull request:
This adds a stage to be able to add kernel arguments on a system by
appending to the BLS [1] config directly either in the tree or in
a mount. This is useful on say systems that don't use `grubby` and
thus can't use the org.osbuild.kernel-cmdline stage.

[1] https://freedesktop.org/wiki/Specifications/BootLoaderSpec/

dustymabe commented:
Naming is hard. I'm open to suggested improvements to the new stage name.

supakeen commented on stages/org.osbuild.kernel-cmdline.bls-append:
```diff
@@ -0,0 +1,71 @@
+    # disk image rather than the tree. If not then use
+    # the tree.
+    mountroot = paths["mounts"]
+    boot = os.path.join(mountroot, "boot")
```
Can this actually be hardcoded to "/boot" or should we communicate this in a different way? I know it's semi-default but my systems don't have a `/boot` in some cases (but only an `/efi`).

achilleas-k replied:
I agree.  Let's add an option for `bls_path` (or simply `path`?) and have it default to `/boot`.

achilleas-k reviewed (commented):
I agree with Simon's comment on adding a path option.  Otherwise LGTM.

dustymabe commented:
ok I added a new commit to I think more comprehensively handle the user specifying a location where the boot root is. I added it as a separate commit for now to make it easier to review. I'll squash it into the previous commit if this looks like the right approach.

achilleas-k commented on stages/org.osbuild.kernel-cmdline.bls-append:
```diff
@@ -0,0 +1,86 @@
+    kopts = options.get("kernel_opts", [])
+    bootpath = options.get("bootpath", "tree:///boot")
+
+    url = urlparse(bootpath)
+    scheme = url.scheme
+    if scheme == "tree":
+        root = tree
+    elif scheme == "mount":
+        root = paths["mounts"]
+    else:
+        raise ValueError(f"Unsupported scheme '{scheme}'")
```
I was looking at the first commit again now and was going to say we should do exactly this.  The caller should be explicit about whether to use the tree or the mount.

Does the `root = paths["mounts"]` work or does it need to get the `"path"` from the sub-object too?
(see https://github.com/osbuild/osbuild/blob/d52738d70c5841613387e6dcd60e1a2f18c56d27/stages/org.osbuild.copy#L90-L99)

dustymabe replied:
> Does the `root = paths["mounts"]` work or does it need to get the `"path"` from the sub-object too?
> Check out the `parse_mount()` function from the copy stage:

I'm not sure. I think I copied the `root = paths["mounts"]` from somewhere else and this seems to work, though I'm not sure if I'm just getting lucky or not.

dustymabe replied:
ok yeah here is where I copied it from:

https://github.com/osbuild/osbuild/blob/cf00c5b21482d572709e9a5353676d028ce587de/stages/org.osbuild.zipl.inst#L90

dustymabe commented:
Done - and also rebased on top of latest.

achilleas-k reviewed (dismissed):
Thanks!!  LGTM

dustymabe commented:
noticed one thing that needed a fixup - did a new push ^^
